### DISCUSSION TITLE
Does it still make sense to run ESLint using the pre-commit hook and manually manage the required dependencies?

### DISCUSSION THREAD

**Discussion by sarayourfriend** [WordPress/openverse#2197] [Q&A] [answered]
Right now we use pre-commit to configure ESLint. This requires us to duplicate our ESLint dependencies (and update the versions in two places) by hand. I believe we originally did this so that it would be easy to run ESLint on our several repositories with some shared baseline configuration. However, now that we have all our JavaScript in the monorepo, does this still make sense? I can see that it is probably nice that ESLint can run without needing to run `pnpm install` locally. However, the hook only runs when JavaScript files are changed, so I think it would be reasonable to expect the JavaScript development dependencies to exist locally.

If that is the case, then we could switch the ESLint hook in our pre-commit configuration to a local hook and run ESLint directly. This would remove the need to manually duplicate the dependencies in two places.

Pinging @dhruvkb specifically for advice regarding the monorepo stuff and because you were foundational in getting the nice pre-commit configuration we have now. Does anything about this stand out to you? Can it be done or would it cause other issues that are better avoided?

**Comment by dhruvkb** [comment]
Using ESLint as a local hook makes sense (maybe even the same for Prettier), and I am all for deduplication anywhere possible. There are other local hooks that run when JS/TS files have changed and so installing the Node.js dependencies is already a requirement at this point.

The only obstacle I can imagine with this is, will we need to install ESLint and its dependencies in all individual workspaces?

**Reply by sarayourfriend** [reply]
Regarding the workspace question: can we just install it in the root workspace? Our ESLint configuration is global and we can port overrides for specific folders up into the root file under `overrides` like I did for the frontend unit tests in https://github.com/WordPress/openverse/pull/2199.

In my experience, other JS monorepos always had stuff like eslint and jest configured at the root level rather than on a per-package basis with packages adding individual overrides when needed rather than having entirely separate configurations. Those dependencies weren't declared in the individual packages either, just in the dev dependencies of the monorepo (unless they were truly package-specific).

**Reply by dhruvkb** [reply, marked as answer]
In that case, this works for me. Let's make an issue out of this discussion and get it done 🚀!

**Reply by sarayourfriend** [reply]
The issue is here: https://github.com/WordPress/openverse/issues/2200
